apps/lib/opt.c

```diff
@@ -899,6 +875,37 @@ void opt_help(const OPTIONS *list)
         }
         start[width] = '\0';
         opt_printf_stderr("%s  %s\n", start, help);
+}
+
+void opt_help(const OPTIONS *list)
+{
+    const OPTIONS *o;
+    int i;
+    int standard_prolog;
+    int width = 5;
+    char start[80 + 1];
+
+    /* Starts with its own help message? */
+    standard_prolog = list[0].name != OPT_HELP_STR;
```

Comment:
Soooo...  this is a bit of a probem.  What if someone, somewhere, has this:

``` C
const OPTIONS foo[] = {
    OPT_SECTION("General"),
    ...
```

How to decide it the resulting OPTIONS item is a section start or a usage?

One quick and fairly easy solution is to have another magic string called `OPT_USAGE_STR` and replace all things looking like this:

``` C
    {OPT_HELP_STR, 1, '-'. "Usage: ..."},
```

with

``` C
    {OPT_USAGE_STR, 1, '-'. "Usage: ..."},
```

Reply:
The other option is to have a magic `OPT_SECTION_STR` instead of re-using `OPT_HELP_STR`.  That's probably going to be less work.

Reply:
Yeah using OPT_SECTION_STR seems like a smaller change that won't break what's already there.